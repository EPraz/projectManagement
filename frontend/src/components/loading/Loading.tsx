import { CircularProgress, Box, Typography } from "@mui/material";

const Loading = ({ message = "Cargando..." }) => {
  return (
    <Box
      display="flex"
      flexDirection="column"
      justifyContent="center"
      alignItems="center"
      height="100vh"
    >
      <CircularProgress size={50} />
      <Typography mt={2} variant="h6" color="textSecondary">
        {message}
      </Typography>
    </Box>
  );
};

export default Loading;
