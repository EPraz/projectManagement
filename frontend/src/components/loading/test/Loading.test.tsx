import { render, screen } from "@testing-library/react";
import Loading from "../Loading";

describe("Loading Component", () => {
  it("match snapshot", async () => {
    const { asFragment } = render(
      <Loading variant="minimal" message="Loading minimal..." />
    );
    expect(asFragment).toMatchSnapshot();
  });

  it("renders minimal variant correctly", () => {
    render(<Loading variant="minimal" message="Loading minimal..." />);
    const progressBar = screen.getByRole("progressbar");
    expect(progressBar).toBeInTheDocument();
    expect(screen.getByText("Loading minimal...")).toBeInTheDocument();
  });

  it("renders default variant correctly", () => {
    render(
      <Loading
        variant="default"
        message="Default Loading"
        description="Please wait while content loads"
      />
    );
    expect(screen.getByText("Default Loading")).toBeInTheDocument();
    expect(
      screen.getByText("Please wait while content loads")
    ).toBeInTheDocument();
    const dots = document.querySelectorAll(".dot");
    expect(dots.length).toBe(3);
  });

  it("renders fullscreen variant with proper height", () => {
    const { container } = render(
      <Loading
        variant="fullscreen"
        message="Fullscreen Loading"
        description="Please wait..."
      />
    );
    const loadingContainer = container.firstChild;
    expect(loadingContainer).toHaveStyle("height: 100vh");
  });
});
