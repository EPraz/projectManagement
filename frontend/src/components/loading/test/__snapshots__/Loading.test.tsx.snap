// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Component match snapshot 1`] = `[Function]`;
